src/main/java/com/google/firebase/auth/UserIdentifier.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.auth;
+
+import com.google.firebase.auth.internal.GetAccountInfoRequest;
+import com.google.firebase.internal.NonNull;
+
+/**
+ * Identifies a user to be looked up.
+ */
+public abstract class UserIdentifier {
```

Comment:
Might be too late for this conversation. But I wonder if we should make all implementations of this class package-protected and provide a set of static factory methods instead.

```
public static UserIdentifier fromUid(String uid);
public static UserIdentifier fromEmail(String email);
// and so on
```

Does that make sense?

Reply:
Discussed offline. tl;dr: for better or worse, we're going to keep it as is for now.